### ISSUE TITLE
Languages, theme configuration error

### ISSUE THREAD

**Issue by GitProQ** [mAAdhaTTah/babel-plugin-prismjs#22]
`[
      'prismjs',
      {
        'languages': ['markup', 'css', 'javascript', 'bash', 'json', 'less', 'php', 'scss', 'sql'],
        'plugins': ['line-numbers', 'show-language', 'highlight-keywords', 'toolbar', 'copy-to-clipboard'],
        'theme': 'solarizedlight',
        'css': true
      }
    ]`
How to use the default theme？
If I use scss，The browser will report an error：`Uncaught TypeError: Cannot set property 'comment' of undefined `

**Comment by BenParizek** [comment]
I'm running into this error as well. I haven't tested with plugins or themes, but the error seems to be specific to certain languages. I had to remove `markup`, `html`, `css`, `javascript`, and `graphql`. Some of those languages render even when removed, like `html` and `css`. However, Javascript doesn't render using the keyword `javascript` or `js` and breaks all other rendering on the page showing that same 'comment` undefined error in the console.

All the syntaxes in the following list are currently working:

```
{
    "plugins": [
        ["prismjs", {
            "languages": [
                "json",
                "php",
                "php-extras",
                "regex",
                "scss",
                "shell",
                "sql",
                "twig",
                "yaml"
            ]
        }]
    ]
}
```

**Comment by BenParizek** [comment]
Ok, I don't think this is a bug, and it's still possible I'm misunderstanding something, but it might just be the documentation that needs to be clarified. I seem to have things working now. Here are a few of my errors, however novice, if it helps anybody else get to the bottom of a similar issue:

1. I don't believe updating the `.bashrc` file and just refreshing actually test any changes to the file. I needed to exist my current watch process and re-run `npm run watch` (the command in my setup) to reinitialize a watch task with the updated changes
2. Upon closer inspection, I noticed that the `javascript` syntax extended the `clike` syntax, which I had omitted from my list. Adding `clike` to the list helped get the `javascript` syntax working.
3. Upon closer inspection, the stack track of the `Uncaught TypeError: Cannot set property 'comment' of undefined` error was referencing the `scss` syntax which extends the `css` syntax. Adding both of those in seems to get things working.

I expect I had a combination of errors as I swapped in various languages without much of a plan, but my guess here is that to avoid this error you need to make sure the 4 primary [supported languages](https://prismjs.com/#supported-languages) are initialized if you are extending them in any way.

My updated, and currently working `.bashrc`: 

```
{
    "plugins": [
        ["prismjs", {
            "languages": [
                "markup",
                "css",
                "clike",
                "javascript",
                "markup-templating",
                "graphql",
                "json",
                "php",
                "php-extras",
                "regex",
                "scss",
                "shell",
                "sql",
                "twig",
                "yaml"
            ]
        }]
    ]
}
```

**Comment by mAAdhaTTah** [comment]
No, it's likely this is a bug on the plugin's side. I was trying to work out the dependency logic and the result was pretty mediocre. We have an implementation to resolve this on Prism's side that I should probably review & get merged in so we can use it here.

**Comment by adam-boduch** [comment]
I have the same bug. my currently working config:

```
{
    "presets": [
        "env"
    ],
    "plugins": [
        [
            "prismjs",
            {
                "languages": [
                    "markup",
                    "css",
                    "clike",
                    "javascript",
                    "markup-templating",
                    "js-extras",
                    "js-templates",
                    "ada",
                    "bash",
                    "basic",
                    "batch",
                    "brainfuck",
                    "c",
                    "csharp",
                    "cpp",
                    "fsharp",
                    "clojure",
                    "diff",
                    "docker",
                    "java",
                    "haskell",
                    "scala"
                ],
                "plugins": [
                    "line-numbers"
                ],
                "css": false
            }
        ]
    ]
}
```

When I add `julia` at the end, code breaks with error 

Config that breaks the code: `Cannot read property 'class-name' of undefined`

```
{
    "presets": [
        "env"
    ],
    "plugins": [
        [
            "prismjs",
            {
                "languages": [
                    "markup",
                    "css",
                    "clike",
                    "javascript",
                    "markup-templating",
                    "js-extras",
                    "js-templates",
                    "ada",
                    "bash",
                    "basic",
                    "batch",
                    "brainfuck",
                    "c",
                    "csharp",
                    "cpp",
                    "fsharp",
                    "clojure",
                    "diff",
                    "docker",
                    "java",
                    "haskell",
                    "scala",

                    "julia"
                ],
                "plugins": [
                    "line-numbers"
                ],
                "css": false
            }
        ]
    ]
}
```

**Comment by jenkoian** [comment]
@adam-boduch I have exact same issue but instead of `julia` the issues occurs when I add `ruby`. It seemed odd, so I swapped ruby out for some other languages and it always seemed to error. What's really odd is if I remove this last line it all seems to work ok.

Working:
<details>
```json
{
    "presets": ["es2015"],
    "plugins": [
        "transform-es2015-modules-commonjs",
        [
            "prismjs",
            {
                "languages": [
                    "c",
                    "clike",
                    "cpp",
                    "csharp",
                    "css",
                    "javascript",
                    "markup",
                    "php",
                    "python"
                ],
                "plugins": [
                    "copy-to-clipboard",
                    "toolbar"
                ],
                "theme": "default",
                "css": true
            }
        ]
    ]
}
```
</details>

Not working:
<details>
```json
{
    "presets": ["es2015"],
    "plugins": [
        "transform-es2015-modules-commonjs",
        [
            "prismjs",
            {
                "languages": [
                    "c",
                    "clike",
                    "cpp",
                    "csharp",
                    "css",
                    "javascript",
                    "markup",
                    "php",
                    "python",
                    "ruby"
                ],
                "plugins": [
                    "copy-to-clipboard",
                    "toolbar"
                ],
                "theme": "default",
                "css": true
            }
        ]
    ]
}
```
</details>

**Comment by mAAdhaTTah** [comment]
Yeaaaaaah my current dep resolution implementation is... not good. We're working on it!

**Comment by jenkoian** [comment]
Just an update on this as I'm digging in, the language in question is a red herring, it seems to be the number of languages, for example if I leave in ruby and remove an earlier language it works just fine. 10 was the tipping point for me (9 works fine) but it seems that other people have way more than that so 🤷‍♂ 

@mAAdhaTTah if you're on top of it I'll stop sharing anything I find, thanks!

**Comment by mAAdhaTTah** [comment]
@jenkoian Yeah, we need to get this landed first: https://github.com/PrismJS/prism/pull/1998 and we'll use that logic in the babel plugin as well. That's a complete rewrite so bugs in the old logic shouldn't transfer. I appreciate you looking into this though!

**Comment by mAAdhaTTah** [comment]
This fix will be released in v2.0.0 (working through some last things now). It's v2.0.0 because you'll need Prism 1.18.0 as your minimum version (rather than 1.0.0) which is breaking to anyone using an older version.
